### DISCUSSION TITLE
Looking for a way to hook up a regular foot switch

### DISCUSSION THREAD

**Discussion by Dalowend** [Builty/TonexOneController#21] [Ideas]
Hi, love this project, thanks!

I started off this discussion on the YT channel but this may be a better place. I’m interested to hook up a regular foot switch (like Boss FS-7) to the Zero module and cycle through the presets. Can you help with providing some code? I am assuming I would just need to assign and connect 2 GPIOs?

**Comment by Builty** [comment]
Hi. I did that coding last week, and its committed to the main branch, but I haven't done any release builds yet (I'm still working on some other parts.)
If you are able to build the code, you could try now, otherwise I need a little more time to do that.
Yes its a as simple as wiring up to GPIO. 
Refer to main.h:
    // direct IO pins
    #define FOOTSWITCH_1		GPIO_NUM_4
    #define FOOTSWITCH_2		GPIO_NUM_6
   
Just need to connect the footswitches to make this 2 input pins become grounded momentarily when the switch is pressed.

**Reply by Dalowend** [reply]
That's helpful, thanks! I can start working on the HW, I'm OK to wait for the SW build. I could try but may need some digging to set things up.. Last time I created some code, Pascal and assembler were still the hype ;-)

**Comment by Dalowend** [comment]
I suppose you may want to add this information here: https://github.com/Builty/TonexOneController/blob/main/HardwarePlatforms.md#hardware-platform-2-waveshare-esp32-s3-zero-no-display-support

**Comment by Builty** [comment]
Yeah sorry haven't quite gotten to updating the docs for this bit yet. I will get it done soon.

**Reply by Dalowend** [reply]
no apologies needed, thanks for the great work!

**Reply by Builty** [reply]
All documentation is to date now.

**Reply by Dalowend** [reply]
thanks!

**Comment by Builty** [comment]
completed
